src/hierarchies/abstract_hierarchy.h

```diff
@@ -265,6 +259,15 @@ class AbstractHierarchy {
       throw std::runtime_error("Not implemented");
     }
   }
+
+  //! Resets summary statistics for this cluster
+  virtual void clear_summary_statistics() = 0;
+
+  //! Initializes state parameters to appropriate values
+  virtual void initialize_state() = 0;
```

Comment:
Do we need these in the `AbstractHierarchy`? I think it would be cleaner to have only the public API here and move these to the `BaseHierarchy`